### PR TITLE
[terra-navigation-side-menu] Accessibility Changes

### PR DESCRIPTION
**Pull request by MadanKumarGovindaswamy**
### Summary

**What was changed:**

Audit Report Issues:

1. Terra must provide a prop allowing consuming teams to provide a label that accurately describes the purpose of the navigation. Add an aria-label to the NAV element and apply consumer provided string.
2. Add dashed border for focused item.
3. Terra Navigation Side Menu that has a caret icon in its sub menu of contrast ratio 1.8:1 that  does not meet the non text contrast ratio of 3:1. Implementing team(s) must ensure that any non-text content that conveys meaning to the user meet the 3:1 contrast ratio.
4. Support Keyboard Navigation(Up and Down Arrow key)

**Why it was changed:**

This component does not commit accessibility criteria.

https://github.com/cerner/terra-framework/assets/132886427/bd69f12d-a9fd-4562-8fd5-8e05ba2b35f2

### Testing

This change was tested using:

- [x] WDIO
- [x] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [x] Accessibility review
- [x] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-10196 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-navigation-side-menu/src/MenuItem.module.scss [hunk 1]
```diff
@@ -107,7 +107,7 @@
       }
 
       .chevron {
-        color: var(--terra-navigation-side-menu-item-selected-chevron-color, #909496);
+        color: var(--terra-navigation-side-menu-item-selected-chevron-color, #1C1F21);
```
Comment (saket2403) [inline packages/terra-navigation-side-menu/src/MenuItem.module.scss:110]: Was this color provided by the design team?

File: packages/terra-navigation-side-menu/src/MenuItem.module.scss [hunk 2]
```diff
@@ -117,17 +117,7 @@
...
-      }
-
-      &.is-focused {
-        background-color: var(--terra-navigation-side-menu-item-selected-focus-background-color, #f4f4f4);
-        color: var(--terra-navigation-side-menu-item-selected-focus-color);
-
-        /* stylelint-disable-next-line max-nesting-depth */
-        .chevron {
-          color: var(--terra-navigation-side-menu-item-selected-focus-chevron-color, #909496);
+          color: var(--terra-navigation-side-menu-item-selected-hover-chevron-color, #1C1F21);
```
Comment (saket2403) [inline packages/terra-navigation-side-menu/src/MenuItem.module.scss:120]: Was this color provided by the design team?

**Comment by saket2403** [comment]
Please add a wdio test to verify the up and down arrow key navigation and also update the test files with the new ariaLabel prop.

File: packages/terra-navigation-side-menu/src/MenuItem.module.scss (hunk 1 above)
Comment (MadanKumarGovindaswamy) [reply packages/terra-navigation-side-menu/src/MenuItem.module.scss]: color contrast change reverted. Thanks
f257333e9538e426d69ce329087fe07892bc97da

File: packages/terra-navigation-side-menu/src/MenuItem.module.scss (hunk 2 above)
Comment (MadanKumarGovindaswamy) [reply packages/terra-navigation-side-menu/src/MenuItem.module.scss]: color contrast change reverted. Thanks
f257333e9538e426d69ce329087fe07892bc97da

File: packages/terra-navigation-side-menu/src/orion-fusion-theme/NavigationSideMenuItem.module.scss [hunk 3]
```diff
@@ -36,16 +36,19 @@
     --terra-navigation-side-menu-item-selected-before-background-image: none;
     --terra-navigation-side-menu-item-selected-background-size: auto;
     --terra-navigation-side-menu-item-selected-border-left: 0.5rem solid #007cc3;
-    --terra-navigation-side-menu-item-selected-chevron-color: #909496;
+    --terra-navigation-side-menu-item-selected-chevron-color: #1C1F21;
```
Comment (saket2403) [inline packages/terra-navigation-side-menu/src/orion-fusion-theme/NavigationSideMenuItem.module.scss:39]: I think you forgot to revert this

File: packages/terra-navigation-side-menu/src/orion-fusion-theme/NavigationSideMenuItem.module.scss [hunk 4]
```diff
@@ -36,16 +36,19 @@
...
     --terra-navigation-side-menu-item-selected-color: #1c1f21;
     --terra-navigation-side-menu-item-selected-focus-background-color: #f4f4f4;
-    --terra-navigation-side-menu-item-selected-focus-chevron-color: #909496;
+    --terra-navigation-side-menu-item-selected-focus-chevron-color: #1C1F21;
```
Comment (saket2403) [inline packages/terra-navigation-side-menu/src/orion-fusion-theme/NavigationSideMenuItem.module.scss:42]: I think you forgot to revert this

File: packages/terra-navigation-side-menu/src/orion-fusion-theme/NavigationSideMenuItem.module.scss (hunk 3 above)
Comment (MadanKumarGovindaswamy) [reply packages/terra-navigation-side-menu/src/orion-fusion-theme/NavigationSideMenuItem.module.scss:39]: yeah. this variable is not used any more. Removed . Thanks
3017b29e248e24b6a7a1746aac6243fccca50af3

File: packages/terra-navigation-side-menu/src/orion-fusion-theme/NavigationSideMenuItem.module.scss (hunk 4 above)
Comment (MadanKumarGovindaswamy) [reply packages/terra-navigation-side-menu/src/orion-fusion-theme/NavigationSideMenuItem.module.scss]: yeah. this variable is not used any more. Removed . Thanks
3017b29e248e24b6a7a1746aac6243fccca50af3

**Review by saket2403 — APPROVED** [review]

File: packages/terra-navigation-side-menu/CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+* Added
+  * New stylings for the focus indicator.
```
Comment (sugan2416) [inline packages/terra-navigation-side-menu/CHANGELOG.md:6]:
```suggestion
  * Added dashed focus border for navigation menu items.
```

**Comment by sugan2416** [comment]
With VO, unable to navigate items with arrow keys, tab works. 
aria-label added in this example: https://github.com/cerner/terra-framework/pull/2018/files#diff-645e1bb824a065f0d02a73ac27b999151d5d51c76124df924b1d19553766de95R63 is not announced.
Can you verify this in both SR

File: packages/terra-navigation-side-menu/src/NavigationSideMenu.jsx [hunk 6]
```diff
@@ -167,6 +174,19 @@ class NavigationSideMenu extends Component {
     }
   }
 
+  handleMenuListRef = (node) => {
+    this.menuContainer = node;
+    // To add focus to the first sub menu item
+    if (node && this.needsFocus) {
+      const subMenuNodes = node.querySelectorAll('[data-menu-item]');
+      if (subMenuNodes) {
```
[the remark this answers is not in the document]
Comment (MadanKumarGovindaswamy) [reply packages/terra-navigation-side-menu/src/NavigationSideMenu.jsx]: Done. Thanks
5382cb3b972bf84832fc30e96516fa9e1f0baba8

**Review by sugan2416 — APPROVED** [review]

**Comment by rahulkumar8cs** [comment]
+1 for the current a11y changes.
We need to fix the focus behaviour navigation side menu of back button in new jira which you created. Thank you

**Pull request merged**